### PR TITLE
Implement RetryBatchSpecExecution

### PR DESCRIPTION
**Pull request by mrnugget**
Part of https://github.com/sourcegraph/sourcegraph/issues/26918 and implements the `RetryBatchSpecExecution` mutation.

### CODE REVIEW

File: enterprise/internal/batches/service/service.go [hunk 1]
```diff
@@ -1270,3 +1272,101 @@ func (s *Service) RetryBatchSpecWorkspaces(ctx context.Context, workspaceIDs []i
...
+
+	var (
+		jobsToDelete           []int64
+		changesetSpecsToDelete []int64
+		workspacesToRetry      []int64
+	)
+
+	for _, j := range jobs {
+		// TODO: We only retry/re-create the jobs that are retryable. What do
+		// we do with others?
```
Comment (mrnugget) [inline enterprise/internal/batches/service/service.go:1341]: @eseliger wdyt?

Comment (eseliger) [reply enterprise/internal/batches/service/service.go:1341]: I think we should leave them untouched. In the EM 3.1 ticket I mentioned we might want to add a flag to just retry _ALL_ if you're waiting for a cache to bust, for example. So maybe we will end up having two paths?

Comment (mrnugget) [reply enterprise/internal/batches/service/service.go:1341]: Makes sense to me. So this will only retry completed/failed, but not queued/processing/etc.

Comment (eseliger) [reply enterprise/internal/batches/service/service.go:1341]: I could be convinced that the right call would be that all need to be in a terminal state. WDYT?

Comment (eseliger) [reply enterprise/internal/batches/service/service.go:1341]: I like that

**Comment by mrnugget** [comment]
@sourcegraph/batchers I'm going to merge this now, since I already looked at this with @eseliger yesterday and we paired on finishing this. But more importantly: there's a lot of PRs open right now that touch the same areas of the code and it helps understanding things if we don't have 4 separate "base branches" to keep in our heads. Of course I'll address any comments post-merge and the UI: well, somebody else needs to make that pretty, sorry!

**Pull request merged**